rfcs/20240311-verbose-logging-support/README.md

```diff
@@ -0,0 +1,52 @@
+
+## Proposal
+The proposal is to implement logging support in oneDNN with the help of [`spdlog`]((https://github.com/gabime/spdlog)), a header-only C++ logging library which provides a Python-like formatting API using the bundled [fmt](https://github.com/fmtlib/fmt) lib. The library uses an MIT license, has no cross-component dependencies and has a distinct performance [gain](https://github.com/gabime/spdlog/tree/v1.x?tab=readme-ov-file#benchmarks) over `sprintf` for logging. Key considerations for the proposal are listed as follows:
+
+### 1. Build Options
+`spdlog` can be built with oneDNN using a header-only approach by adding the library headers to the build tree. A pre-compiled version is recommended by the authors to reduce compilation overhead but is not required since the implementation will utilize only limited features from the entire scope of `spdlog`. 
+
+### 2. Runtime Logging Controls
+A basic requirement for implementing logging support will be to define the environmental control variables which the user can specify to manage oneDNN data logging. 
+For the simple case where the logging mechanism involves directly dumping the verbose outputs into a logfile, this can be accomplished with two control variables, one for enabling the data logging (`ENABLE_ONEDNN_LOGGING`) and the other for specifying the logfile path (`ONEDNN_LOGFILE=/path/to/file`).
+In this scenario, the data recorded in the logfile mirrors the printed verbose mode information, hence, the logged data can be managed using oneDNN [runtime controls](https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html?highlight=onednn_verbose#run-time-controls) for the verbose mode.
+For better logging capabilities, additional runtime controls can be introduced to separately control the logging levels for the saved data. 
+
+### 3. Alignment and Specification of Logging Levels
+`spdlog` defines the following levels for data logging in its implementation:
+```
+#define SPDLOG_LEVEL_TRACE 0
+#define SPDLOG_LEVEL_DEBUG 1
+#define SPDLOG_LEVEL_INFO 2
+#define SPDLOG_LEVEL_WARN 3
+#define SPDLOG_LEVEL_ERROR 4
+#define SPDLOG_LEVEL_CRITICAL 5
+#define SPDLOG_LEVEL_OFF 6 
+```
+
+The type of tracing information logged for each of these levels is evident from their names. Comparing with the different [verbose modes](https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html?highlight=onednn_verbose#run-time-controls) defined for oneDNN, these logging levels can be aligned with the verbose modes as follows:
+
+|   | `spdlog` Level          | oneDNN Verbose Mode                                              |
+|---|-----------------------|------------------------------------------------------------------|
+| 0 | SPDLOG_LEVEL_TRACE    | ---                                                              |
+| 1 | SPDLOG_LEVEL_DEBUG    | `dispatch`, `debuginfo=<level>`                                  |
```

Comment:
`check` should be added in this level.

Reply:
Thanks for the correction! I have changed the logging level for `check`.